package.json

```diff
@@ -20,7 +20,11 @@
     "@azure/avocado": "^0.8.4",
     "@types/prettier": "^2.7.2",
     "prettier": "^2.8.8",
-    "typescript": "~5.0.4"
+    "typescript": "~5.0.4",
+    "typespec-validation": "file:eng/tools/TypeSpecValidation"
+  },
+  "scripts": {
+    "postinstall": "npx --no tsc -- -p eng/tools/TypeSpecValidation"
```

Comment:
Should this be a postinstall step of the TypeSpecValidation package.json instead? I'm not sure if that is the standard strategy but it could work for us, esspecially if we end up with more tools we need to compile.

Reply:
The TypeSpecValidation package itself should not have a "postinstall" step, since if you were consuming the pre-compiled package from a tarball or a feed, you wouldn't want any post-install steps.  So I think it's best to put the "postinstall" step in the same place that depends on the package directly from TypeSpec sources.

Reply:
Regarding the unified pipeline, I believe the problem was running `tsc` in the repo root, which was trying to compile too much stuff.  It should be safe to only compile folders known to not be touched by the unified pipeline.  And the unified pipeline doesn't seem to be impacted in this PR itself.